### ISSUE TITLE
File upload component: improve hover and cursor affordances

### ISSUE THREAD

**Issue by rickybanister** [Automattic/wp-calypso#44319]
![image](https://user-images.githubusercontent.com/437258/88087960-b192c280-cb57-11ea-9cb6-81508f63cd31.png)

The drag area for our file upload also mentioned clicking to select a file. 

There is no hover state and the cursor does not change to make that more obvious. The cursor actually changes from default to text as you scroll through the box. It should be pointer or perhaps other patterns are prevalent for this type of UI. 

We should also look into any similar @wordpress/component work.

**Comment by sixhours** [comment]
Researching this, and I'll drop my findings here. We use a few custom implementations of the DropZone component which accounts for this discrepancy.

The Imports section appears to be using [DropZone with a custom onClick wrapper](https://github.com/Automattic/wp-calypso/blob/15a8bd965e6ababd85f8869f989fa4e1b1d1f139/client/my-sites/importer/uploading-pane.jsx):

<img width="773" alt="Screen Shot 2020-10-05 at 11 57 36 AM" src="https://user-images.githubusercontent.com/2124984/95102975-0af3a200-0702-11eb-8109-4d3a642f46d4.png">

The Gravatar changer also uses a custom implementation:

<img width="272" alt="Screen Shot 2020-10-05 at 11 56 50 AM" src="https://user-images.githubusercontent.com/2124984/95102868-e992b600-0701-11eb-9cc0-ff1209bef2d9.png">

Whereas plugins and themes use the [`UploadDropZone` block](https://github.com/Automattic/wp-calypso/blob/4537033c6794449ce0a10b69d72daf5fa658d553/client/blocks/upload-drop-zone/index.jsx), which has [built-in hover and cursor styles](https://github.com/Automattic/wp-calypso/blob/4537033c6794449ce0a10b69d72daf5fa658d553/client/blocks/upload-drop-zone/style.scss#L6):

<img width="766" alt="Screen Shot 2020-10-05 at 11 57 51 AM" src="https://user-images.githubusercontent.com/2124984/95102963-06c78480-0702-11eb-9a74-c7f6a79a178d.png">

We also have an [`ImageSelector` block](http://calypso.localhost:3000/devdocs/blocks/image-selector), which interfaces with the Media Library to upload/add images only:

<img width="403" alt="Screen Shot 2020-10-05 at 12 00 33 PM" src="https://user-images.githubusercontent.com/2124984/95103436-98cf8d00-0702-11eb-97a3-a1a7d2548ffd.png">

At first glance, replacing the custom `DropZone` implementations with the `UploadDropZone` block *might* work.

Taking a step further, we could style the `UploadDropZone` block to look and work more like core's, with a specific call to action in the drop zone, rather than clicking the whole area:

<img width="815" alt="Screen Shot 2020-10-05 at 11 47 55 AM" src="https://user-images.githubusercontent.com/2124984/95102115-fa8ef780-0700-11eb-857b-b0ebdc9e77e8.png">

Will keep looking to see if there are additional implementations in core.

**Comment by sixhours** [comment]
Core's Featured Image component has a very simple hover effect on the background color:

https://cloudup.com/c63HLmrPYO5

**Comment by sixhours** [comment]
The core media library has a similar setup to the block editor's Image/Gallery selectors:

<img width="1511" alt="Screen Shot 2020-10-05 at 1 45 53 PM" src="https://user-images.githubusercontent.com/2124984/95113925-2c0fbf00-0711-11eb-8aef-5219cc65fe80.png">
